tests/test_kubernetes_tools.py

```diff
@@ -1216,6 +1216,36 @@ def test_get_volume_mounts(self):
                 == expected_volumes
             )
 
+    @pytest.mark.parametrize(
+        "config_dict",
+        [
+            {"boto_keys": ["scribereader"]},
+            {"boto_keys": ["scribereader", "foobar"]},
+            {},
+        ],
+    )
+    def test_get_boto_volume(self, config_dict):
+        deployment = KubernetesDeploymentConfig(
+            service="my-service",
+            instance="my-instance",
+            cluster="mega-cluster",
+            config_dict=config_dict,
+            branch_dict=None,
+            soa_dir="/nail/blah",
+        )
+        with mock.patch.object(deployment, "get_boto_secret_hash", return_value="hash"):
+            volumes = deployment.get_boto_volume()
+        if config_dict:
+            assert (
+                volumes.secret.secret_name == "paasta-boto-key-my-service-my-instance"
+            )
+            assert len(volumes.secret.items) == len(config_dict["boto_keys"]) * 4
+            assert any(
+                [item.path == "scribereader.yaml" for item in volumes.secret.items]
```

Comment:
should this look at config_dict to figure out what to check for? technically we could break things such that we only use the path for the first item in the boto_keys list and this test would still pass :p